com.spotify/index.js

```diff
@@ -0,0 +1,662 @@
+      if(albumResult.albums.total == 0) return;
+      albumResult = albumResult.albums.items[0].uri;
+      console.log(albumResult);
+      let playlistResult = await this.find_my_playlist(playlist);
+      console.log(playlistResult);
+      if (playlistResult.length == 0) return;
+      playlistResult = playlistResult.substring(playlistResult.indexOf("playlist:") + 9);
+      let albumTracks = await this.get_album_tracks(albumResult);
+      console.log("Album tracks are " + JSON.stringify(albumTracks));
+      let data = {"uris":albumTracks};
+      return this.add_uris_to_playlist(playlistResult,data);
+    }
+
+    async do_add_multiple_songs_to_playlist({toAdd,playlist}){    //TODO: rename this later to _ "add songs ... "
+      let songs = this.splitMultiString(toAdd);
+      let uris = await this.songNamesToURIs(songs);
+      let playListURI = await this.find_my_playlist(playlist);
+      playListURI = playListURI.substring(playListURI.indexOf("playlist:") + 9);
+      return this.add_uris_to_playlist(playListURI,uris);
+    }
+
+    async do_add_this_song_to_playlist({playlist}){
+      let song = await this.currently_playing_helper();
+      song = JSON.parse(song);
+      let data = {"uris":[song.item.uri]};
+      let playListURI = await this.find_my_playlist(playlist);
+      playListURI = playListURI.substring(playListURI.indexOf("playlist:") + 9);
+      return this.add_uris_to_playlist(playListURI,data);
+    }
+
+    async do_create_new_playlist({title,description,isPublic}){
+      let url = USER_PLAYLISTS;
+      url = url.replace('{username}',this.state.userId);
+      let data = {name:title};
+      if (description) data.description = description;
+      if (isPublic != 'undefined') data.public = isPublic;
+      else data.public = false;
+      return this.http_post_default_options(url,JSON.stringify(data));
+    }
+
+    /*Copyright (c) 2011 Andrei Mackenzie
+    Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+    The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.*/
```

Comment:
Please wrap the license to make it readable. Also this should probably be a function, not a method, and maybe defined in a separate file to keep the licensing cleaner.